### PR TITLE
fix(storefront): BCTHEME-347 Add unique identifiers to product cards

### PR DESCRIPTION
**Pull request by bc-alexsaiannyi**
#### What?

This PR added unique identifiers  on product cards to make a search of specific product card more convenient.

#### Tickets / Documentation

- [BCTHEME-347](https://jira.bigcommerce.com/browse/BCTHEME-347)

#### Screenshots (if appropriate)
data-test attr

### CODE REVIEW

**Comment by BC-tymurbiedukhin** [comment]
This class won't be used in our JS code, should we use data-test-* attribute here?

**Comment by bc-alexsaiannyi** [comment]
> This class won't be used in our JS code, should we use data-test-* attribute here?

yes, it makes sense. Done.

File: templates/components/products/card.html [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<article class="card {{#if alternate}}card--alternate{{/if}}" {{#if settings.data_tag_enabled}} data-event-type="{{event}}" data-entity-id="{{id}}" data-position="{{position}}" data-name="{{name}}" data-product-category="{{#each category}}{{#if @last}}{{this}}{{else}}{{this}}, {{/if}}{{/each}}" data-product-brand="{{brand.name}}" data-product-price="{{#if price.with_tax}}{{price.with_tax.value}}{{else}}{{price.without_tax.value}}{{/if}}"  {{/if}}>
+<article class="card {{#if alternate}}card--alternate{{/if}}" data-test="card-{{id}}" {{#if settings.data_tag_enabled}} data-event-type="{{event}}" data-entity-id="{{id}}" data-position="{{position}}" data-name="{{name}}" data-product-category="{{#each category}}{{#if @last}}{{this}}{{else}}{{this}}, {{/if}}{{/each}}" data-product-brand="{{brand.name}}" data-product-price="{{#if price.with_tax}}{{price.with_tax.value}}{{else}}{{price.without_tax.value}}{{/if}}"  {{/if}}>
```
Comment (BC-tymurbiedukhin) [inline templates/components/products/card.html:1]: Could we pretify attribute list?  There are nested if cases and it takes extra involvement

**Review by BC-tymurbiedukhin — APPROVED** [review]
LGTM

**Review by yurytut1993 — APPROVED** [review]

**Pull request merged**